src/Take.Elephant/ISetMap.cs

```diff
@@ -18,5 +18,7 @@ public interface ISetMap<TKey, TItem> : IMap<TKey, ISet<TItem>>
         /// <param name="cancellationToken"></param>
         /// <returns>An existing set if the key exists; otherwise, an empty set.</returns>
         Task<ISet<TItem>> GetValueOrEmptyAsync(TKey key, CancellationToken cancellationToken = default);
+
+        bool SupportsEmptySets { get; }
```

Comment:
I don't think that having a property with metadata information about the implementations makes sense.

Reply:
We could have a marker interface, such as `IEmptySetSupportingSetMap`, though honestly I think that's only marginally better than the current design. Even though I do think this does qualify it as "good", I think it's worth pointing out that things like this are certainly not unheard of; See "IsReadOnly" on the BCL `IList<T>` interface or `CanSeek` on the `Stream` class.

Anyhow, we would appreciate a suggestion of an alternative design; We really did scratch our heads with this one and this is what we ended up choosing.

Reply:
A suggestion would be creating a distinct interface, like `IEmptySetMap<TKey, TItem>`. You could move the `GetValueOrEmptyAsync` value to this interface since it would not make much sense to have this method in a SetMap that doesn't support empty sets.

The point is that the premise in Elephant is that we use interfaces to describe what each storage engine supports or not. So in this case this property do not fit well in the current philosophy adopted by the lib.